core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java

```diff
@@ -4,18 +4,30 @@
 import com.scalar.db.api.TableMetadata;
 import com.scalar.db.exception.storage.ExecutionException;
 import com.scalar.db.io.DataType;
+import com.scalar.db.io.DateColumn;
+import com.scalar.db.io.TimeColumn;
+import com.scalar.db.io.TimestampColumn;
+import com.scalar.db.io.TimestampTZColumn;
 import com.scalar.db.storage.jdbc.query.SelectQuery;
 import com.scalar.db.storage.jdbc.query.UpsertQuery;
 import java.sql.Driver;
 import java.sql.JDBCType;
+import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Collections;
+import java.util.Map;
 import javax.annotation.Nullable;
 
 /**
  * An interface to hide the difference between underlying JDBC SQL engines in SQL dialects, error
  * codes, and so on. It's NOT responsible for actually connecting to underlying engines.
  */
-public interface RdbEngineStrategy {
+public interface RdbEngineStrategy<T_DATE, T_TIME, T_TIMESTAMP, T_TIMESTAMPTZ> {
```

Comment:
I don't have a strong opinion, but having four time-related generics in this interface might be too much since this handles many other data types.
How about pushing the time-related type things to other interface and classes?
For example, how about creating a time-specific strategy that converts the default type to a DB-specific type?
For example, we can create `RdbEngineTimeTypeStrategy`, and it converts time-related types as necessary as follows.

```
preparedStatement.setObject(index++, rdbEngineTimeType.convert(rdbEngine.encodeTimestampTZ(column)));
```

Reply:
It sounds good, thank you.
I reverted an earlier commit based on @komamitsu suggestion in https://github.com/scalar-labs/scalardb/pull/2437/commits/0eee567e6d4385413cb6d1f27f2e37698fc3864b than added your suggestion in 
https://github.com/scalar-labs/scalardb/pull/2437/commits/fe3be8d38000b0f8c7ffc62c27202c62a5bdf4c4